cmd/lvmd/app/root.go

```diff
@@ -108,13 +111,30 @@ func subMain(ctx context.Context) error {
 	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGTERM)
 	defer stop()
 
+	var pprofServer *http.Server
+	if config.ProfilingPort > 0 {
+		go func() {
+			pprofServer = profiling.NewProfilingServer(config.ProfilingPort)
```

Comment:
The assignment should be out of the go statement because the shutdown code might access `pprofServer` before this.